registry/txt.go

```diff
@@ -464,3 +474,19 @@ func (im *TXTRegistry) removeFromCache(ep *endpoint.Endpoint) {
 		}
 	}
 }
+
+func createKey(elem ...string) string {
+	var sb strings.Builder
+
+	for _, e := range elem {
+		if e == "" {
+			continue
```

Comment:
A key for a new-format record for a CNAME "example.com" and no set identifier will collide with a key for an old-format record for "example.com" and a set identifier of `cname`.